### PR TITLE
[llvm] [aot] Added CGraph tests for LLVM backend

### PR DESCRIPTION
**Pull request by jim19930609**
Related issue = #4800

1. Modified RuntimeContext construction inside `CompiledGraph::run()`, so that LlvmAotModuleLoader is able to setup `LLVMRuntime runtime_` upon CompiledGraph construction.

2. Rearranged file structure under /cpp/tests/aot, so that "mpm88" test is shared between llvm and vulkan backends. 

3. Added `tests/cpp/aot/llvm/mpm88_graph_aot_test.cpp`, `tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py`, `tests/cpp/aot/llvm/mpm88_graph_aot_cuda.py` as LLVM AOT test cases.

*Will replace "LlvmProgramImpl" with "LlvmRuntimeExecutor" across all the LLVM AOT tests in a following PR

### CODE REVIEW

File: taichi/program/ndarray.cpp [hunk 1]
```diff
@@ -194,6 +194,14 @@ void set_runtime_ctx_ndarray(RuntimeContext *ctx,
                              int arg_id,
                              Ndarray *ndarray) {
   ctx->set_arg_devalloc(arg_id, ndarray->ndarray_alloc_, ndarray->shape);
+
+  int total_array_size = 1;
```
Comment (ailzhang) [inline taichi/program/ndarray.cpp:198]:
```suggestion
  uint64 total_array_size = 1;
```

File: tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import os
+import sys
+
+import taichi as ti
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), ".."))
```
Comment (ailzhang) [inline tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py:6]: nit: I'm personally not a big fan of manipulating `sys.path`. I'm not too worried about this particular use case but in general afraid of changing `sys.path` spreading across the codebase. 
Can we try passing in arch information to the script?

**Review by ailzhang — APPROVED** [review]
LGTM!!

File: tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py (hunk 2 above)
Comment (jim19930609) [reply tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py:6]: Thanks! Switched to pass "PYTHONPATH" as environmental variables during subprocess call (at tests/run_tests.py).

File: taichi/program/ndarray.cpp (hunk 1 above)
Comment (jim19930609) [reply taichi/program/ndarray.cpp]: Done, thanks!

File: tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py (hunk 2 above)
Comment (jim19930609) [reply tests/cpp/aot/llvm/mpm88_graph_aot_cpu.py]: Thanks! Switched to pass in arguments~

**Review by ailzhang — APPROVED** [review]
Thanks!!

**Pull request merged**